### PR TITLE
[ADD][8.0] sale_service_project: New module.

### PR DESCRIPTION
**Pull request by sergio-teruel**
Create project and tasks with works and materials from sale orders

### CODE REVIEW

File: sale_service_project/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        'data/sale_service_project_demo.xml',
+    ],
+    'data': [
+        'views/sale_service_view.xml',
+        'views/sale_service_project_view.xml',
+        'views/sale_view.xml',
+        'wizard/product_price_service_view.xml',
+    ],
+    'author': 'Incaser Informatica S.L., '
+              'Antiun Ingeniería S.L., '
```
Comment (rafaelbn) [inline sale_service_project/__openerp__.py]: Change order

File: sale_service_project/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+              'Odoo Community Association (OCA)',
+    'website': 'http://www.incaser.es',
+    'license': 'AGPL-3',
```
Comment (rafaelbn) [inline sale_service_project/__openerp__.py:33]: www.antiun.com

**Comment by rafaelbn** [comment]
```
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119405-226-9cfbaf/openerp/addons/base/module/module.py", line 393, in state_update
    raise orm.except_orm(_('Error'), _("You try to install module '%s' that depends on module '%s'.\nBut the latter module is not available in your system.") % (module.name, dep.name,))
except_orm: ('Error', u"You try to install module 'sale_service_project' that depends on module 'project_task_materials'.\nBut the latter module is not available in your system.")
```

**Comment by rafaelbn** [comment]
Please rebase @sergio-incaser

**Comment by rafaelbn** [comment]
ping @sergio-incaser  please rebase for testing, thanks

**Comment by rafaelbn** [comment]
Please @sergio-incaser , rebase and review if runbot fails, thanks!

File: sale_service_project/README.rst [hunk 3]
```diff
@@ -0,0 +1,75 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+
+====================
+Sale Service Project
+====================
+
+This module was written to extends the functionality of sale workflow
```
Comment (yajo) [inline sale_service_project/README.rst]: "This module extends..."

File: sale_service_project/README.rst [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+Usage
+=====
+
+To use this module, you need to:
+
+1. Create a service product and check 'Create Task Automatically'.
+2. Now you can add task works and materials for this product.
+3. You can compute the total price based in all materials and time.
+4. If you want invoice from analytic lines you need select analytic
+order policy option.
```
Comment (yajo) [inline sale_service_project/README.rst]: Prepend 2 spaces to this line.

File: sale_service_project/README.rst [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+Usage
+=====
+
+To use this module, you need to:
+
+1. Create a service product and check 'Create Task Automatically'.
+2. Now you can add task works and materials for this product.
+3. You can compute the total price based in all materials and time.
+4. If you want invoice from analytic lines you need select analytic
```
Comment (yajo) [inline sale_service_project/README.rst]: "... want to invoice..."

File: sale_service_project/README.rst [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+====================
+Sale Service Project
+====================
+
+This module was written to extends the functionality of sale workflow
+and allow you to create projects and tasks with materials and works
+from sale lines.
+You can invoice from sale order or analytic lines.
+
+Configuration
```
Comment (yajo) [inline sale_service_project/README.rst]: Remove this section if empty.

File: sale_service_project/README.rst [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+* https://www.odoo.com/forum/help-1
+
+Known issues / Roadmap
+======================
+
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/167/issues>`_.
```
Comment (yajo) [inline sale_service_project/README.rst]: This repo is sale-workflow, not 167.

File: sale_service_project/README.rst [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+This module is maintained by the OCA.
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+To contribute to this module, please visit http://odoo-community.org.
```
Comment (yajo) [inline sale_service_project/README.rst]: https

File: sale_service_project/models/product.py [hunk 9]
```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 Antiun Ingeniería S.L. - Sergio Teruel
+# (c) 2015 Antiun Ingeniería S.L. - Carlos Dauden
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import models, fields, api, _
+from openerp.addons.decimal_precision import decimal_precision as dp
+
+
+class ProductProduct(models.Model):
```
Comment (yajo) [inline sale_service_project/models/product.py]: ProductTemplate

File: sale_service_project/tests/test_sale_service_project.py [hunk 10]
```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 Antiun Ingeniería S.L. - Sergio Teruel
+# (c) 2015 Antiun Ingeniería S.L. - Carlos Dauden
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp.tests.common import TransactionCase
+
+
+class TestSaleServiceProject(TransactionCase):
+    # Use case : Prepare some data for current test case
```
Comment (yajo) [inline sale_service_project/tests/test_sale_service_project.py]: Use a `"""docstring"""`.

File: sale_service_project/wizard/product_price_service.py [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+        total_price_materials = sum(
+            [m.quantity * m.material_id.lst_price for m in materials])
+        total_price = total_price_hours + total_price_materials
+        return total_price
+
+    @api.multi
+    def action_compute_price(self):
+        product = self.env['product.template'].browse(
+            self._context['active_id'])
+        product.write(
```
Comment (yajo) [inline sale_service_project/wizard/product_price_service.py:32]: You click "Calculate price" from the service product, and the wizard asks you again to select the product. Nonsense.

But anyway, using a wizard for this does not seem to make sense by itself. This should be done in the product model, unless I'm missing something.

File: sale_service_project/README.rst [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+
+To use this module, you need to:
+
+Usage
+=====
+
+To use this module, you need to:
+
+1. Create a service product and check 'Create Task Automatically'.
+2. Now you can add task works and materials for this product.
```
Comment (yajo) [inline sale_service_project/README.rst]: These usage instructions definitely need more precision:
- where to go to make each action?
- what new things you will see in the product, sale order, sale order line, project task...?
- what do they mean?
- what would happen if you add works for tasks in the product? Do the hours mean they are the expected time for task, or are fulfilled time for task works?

I had to guess most of those, so please explain usage better.

**Comment by yajo** [comment]
Tested in localhost, it seems to work fine, but please fix those comments above.

**Comment by rafaelbn** [comment]
As commented today we must fix:
- contract in sale order is parent of new proyect with task of sale order
- in the analytic account associated with the proyect of sale order there is no cost of material, onyl units

**Comment by rafaelbn** [comment]
Hi @carlos-incaser @sergio-incaser runbot fails

```
    openerp.service.server:942 preload_registries

Failed to initialize database `3137323-226-8be756-all`.
Traceback (most recent call last):
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3137323-226-8be756/openerp/service/server.py", line 929, in preload_registries
    registry = RegistryManager.new(dbname, update_module=update_module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3137323-226-8be756/openerp/modules/registry.py", line 370, in new
    openerp.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3137323-226-8be756/openerp/modules/loading.py", line 319, in load_modules
    modobj.button_install(cr, SUPERUSER_ID, ids)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3137323-226-8be756/openerp/api.py", line 256, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3137323-226-8be756/openerp/addons/base/module/module.py", line 414, in button_install
    self.state_update(cr, uid, ids, 'to install', ['uninstalled'], context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3137323-226-8be756/openerp/api.py", line 256, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3137323-226-8be756/openerp/api.py", line 387, in old_api
    result = method(recs, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3137323-226-8be756/openerp/addons/base/module/module.py", line 393, in state_update
    raise orm.except_orm(_('Error'), _("You try to install module '%s' that depends on module '%s'.\nBut the latter module is not available in your system.") % (module.name, dep.name,))
except_orm: ('Error', u"You try to install module 'sale_service_project' that depends on module 'project_task_materials'.\nBut the latter module is not available in your system.")
```

**Comment by rafaelbn** [comment]
Runbot fails

```
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3137566-226-64e02b/openerp/addons/base/module/module.py", line 393, in state_update
    raise orm.except_orm(_('Error'), _("You try to install module '%s' that depends on module '%s'.\nBut the latter module is not available in your system.") % (module.name, dep.name,))
except_orm: ('Error', u"You try to install module 'account_move_batch_validate' that depends on module 'connector'.\nBut the latter module is not available in your system.")
```

Why do you need module account_move_batch_validate?

cc @Yajo @pedrobaeza

**Comment by pedrobaeza** [comment]
Check pylint errors:

```
sale_service_project/models/sale_service.py:6:1: F401 '_' imported but unused
sale_service_project/models/sale_service.py:18:80: E501 line too long (96 > 79 characters)
sale_service_project/models/sale_service.py:19:80: E501 line too long (137 > 79 characters)
```

**Comment by rafaelbn** [comment]
- View task button must be shown only after sale order is confirmed

**Comment by carlosdauden** [comment]
Review @rafaelbn

**Comment by sergio-teruel** [comment]
Changes done.
@rafaelbn @pedrobaeza

File: sale_service_project/README.rst [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+3. You can compute the total price based in all materials and time.
+4. If you want to invoice from analytic lines you need select analytic
```
Comment (pedrobaeza) [inline sale_service_project/README.rst]: This is outdated.

File: sale_service_project/README.rst [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+You can invoice from sale order or analytic lines.
+
+Usage
+=====
+
+To use this module, you need to:
+
+1. Create a service product and check 'Create Task Automatically'.
+2. Now you can add task works and materials for this product.
+3. You can compute the total price based in all materials and time.
```
Comment (pedrobaeza) [inline sale_service_project/README.rst]: This is not instructions for this module.

File: sale_service_project/README.rst [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/167/8.0
+
+For further information, please visit:
+
+* https://www.odoo.com/forum/help-1
+
+Known issues / Roadmap
```
Comment (pedrobaeza) [inline sale_service_project/README.rst]: Remove this empty section

File: sale_service_project/models/hr_timesheet_invoice.py [hunk 16]
```diff
@@ -0,0 +1,108 @@
...
+    def _get_sale_lines(self):
+        timesheet_obj = self.env['hr.analytic.timesheet']
+        timesheet = timesheet_obj.search([('line_id', 'in', self.ids)])
+        task_works = self.env['project.task.work'].search(
+            [('hr_analytic_timesheet_id', 'in', timesheet.ids)])
+        return task_works.mapped('task_id.sale_line_id')
+
+    @api.multi
+    def invoice_cost_create(self, data=None):
+        invoice_obj = self.env['account.invoice']
```
Comment (pedrobaeza) [inline sale_service_project/models/hr_timesheet_invoice.py]: For what is this code needed?

File: sale_service_project/models/sale.py [hunk 17]
```diff
@@ -0,0 +1,140 @@
...
+        for order in self:
+            self.task_ids = self.env['project.task'].search(
+                [('sale_line_id', 'in', order.order_line.ids)])
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form', toolbar=False,
+                        submenu=False):
+        res = super(SaleOrder, self).fields_view_get(
+            view_id, view_type, toolbar=toolbar, submenu=submenu)
+        if view_type == 'form' and not res['fields'].get(
```
Comment (pedrobaeza) [inline sale_service_project/models/sale.py]: Remove this outdated code

File: sale_service_project/README.rst (hunk 14 above)
Comment (sergio-teruel) [inline sale_service_project/README.rst]: Can you explain it to me?

Comment (pedrobaeza) [inline sale_service_project/README.rst]: Sorry, I misunderstood with hr_timesheet_invoice. Anyway, you should indicate that on each product, you can add materials and operations.

**Comment by rafaelbn** [comment]
@carlos-incaser please review https://github.com/OCA/sale-workflow/pull/226#issuecomment-165800149

Now there is not button "view task" neither before or after confirming sale order

captura

**Comment by rafaelbn** [comment]
Also please check travis

File: sale_service_project/wizard/product_price_service.py (hunk 11 above)
Comment (sergio-teruel) [inline sale_service_project/wizard/product_price_service.py:32]: The required product by wizard is used to compute the hours price. We can have distinct price depends of work's complexity.

**Comment by rafaelbn** [comment]
Please review travis, it's failing.

**Comment by yajo** [comment]
Please fix at least Travis and https://github.com/OCA/sale-workflow/pull/226#discussion_r47236325 to keep reviewing.

**Comment by carlosdauden** [comment]
@rafaelbn @Yajo @sergio-incaser Please review

To use this module, you need to:
1.Create a service product and check 'Create Task Automatically'.
2.Now you can add task works and materials for this product.
3.You can compute the total price based in all materials and task works
  assigned.
4.Create sale order and assign an analytic account which will be parent of new
  project.
  If _invoice_on_timesheets_ option is checked on analiytic account this sale
  order will be invoiced from analytic lines. Otherwise sale order will be
  invoiced normally.
5.When you confirm the sale order odoo will create a project and tasks.

**Comment by JavierIniesta** [comment]
Tested in runbot :+1:

**Comment by JMODaERP** [comment]
I´ve tested in runbot :+1:

File: sale_service_project/README.rst [hunk 18]
```diff
@@ -0,0 +1,72 @@
...
+1.Create a service product and check 'Create Task Automatically'.
+2.Now you can add task works and materials for this product.
+3.You can compute the total price based in all materials and task works
+  assigned.
+4.Create sale order and assign an analytic account which will be parent of new
+  project.
+  If *invoice_on_timesheets* option is checked on analiytic account this sale
+  order will be invoiced from analytic lines. Otherwise sale order will be
+  invoiced normally.
+5.When you confirm the sale order odoo will create a project and tasks.
```
Comment (yajo) [inline sale_service_project/README.rst]: Please give [proper formatting](http://docutils.sourceforge.net/docs/user/rst/quickref.html#enumerated-lists) to lists, or they appear broken.

File: sale_service_project/wizard/product_price_service_view.xml [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+        
+        <record id="wiz_compute_task_price" model="ir.ui.view">
+            <field name="name">product.price.service.wizard.form</field>
+            <field name="model">product.price.service.wizard</field>
+            <field name="type">form</field>
+            <field name="arch" type="xml">
+                <form string="Compute Price">
+                        <DIV>
+                            <h3>Select a service product to compute price</h3>
+                        </DIV>
```
Comment (yajo) [inline sale_service_project/wizard/product_price_service_view.xml]: to lower case

Comment (yajo) [inline sale_service_project/wizard/product_price_service_view.xml]: Or, even better, add a `string=` attribute to below `<group>`

**Comment by yajo** [comment]
I have tested in runbot. I did this:
1. Create a service product. Call it "test service".
2. mark create task autom.
3. choose a project.
4. Set 2 hours of work with some name.
5. Set 2 _[A1090] iMac_ in materials.

Then I go to compute price and I get these results:
- I choose no product: 3,598.00
- Run it again, choose no product: 3,598.00
- Run it again, choose "test service": 10,794.00
- Run it again, choose "test service": 25,186.00
- I set price to 0.
- Run it again, choose "test service": 3,598.00

What is this doing? Is that expected? Am I suposed to select own product or leave blank?

In case this behavior is expected, this wizard should definitely be more user-friendly and self-explanatory. Otherwise, it also needs fixing.

**Comment by carlosdauden** [comment]
https://github.com/OCA/sale-workflow/pull/226#discussion_r48236685
@Yajo The product selected in the wizard is the product which the hour's price will be taked . e.g. service, peon hour, official hour, etc.

**Comment by yajo** [comment]
Do you mean that if I set 3 hours and choose iMac as product, the price will be 3 x iMac's price?

**Comment by carlosdauden** [comment]
Yes.
But you only can select service type products.

File: sale_service_project/wizard/product_price_service.py [hunk 20]
```diff
@@ -0,0 +1,30 @@
...
+
+
+class ProductPriceServiceWizard(models.TransientModel):
+    _name = 'product.price.service.wizard'
+
+    product_id = fields.Many2one(
+        comodel_name='product.product', string='Product')
+    uom_id = fields.Many2one('product.uom', string='Unit of measure')
+
+    def _compute_price(self, product, product_service):
```
Comment (yajo) [inline sale_service_project/wizard/product_price_service.py:19]: Either add `@api.model` or replace `product_service` with `self.product_id`.

File: sale_service_project/wizard/product_price_service_view.xml [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+                        <group>
+                            <group>
+                                <field name="product_id"
+                                       domain="[('type','=','service')]"/>
```
Comment (yajo) [inline sale_service_project/wizard/product_price_service_view.xml]: Why not put the domain in the model?

**Comment by yajo** [comment]
OK I understand now, but then definitely the wizard needs some self-explanation.

Also, what happens when you select no product?

File: sale_service_project/wizard/product_price_service_view.xml [hunk 22]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+    <data>
+        
+        <record id="wiz_compute_task_price" model="ir.ui.view">
+            <field name="name">product.price.service.wizard.form</field>
+            <field name="model">product.price.service.wizard</field>
+            <field name="type">form</field>
```
Comment (oihane) [inline sale_service_project/wizard/product_price_service_view.xml]: In v8 you do not have to put the type. It takes it from inside arch.

**Comment by sergio-teruel** [comment]
@rafaelbn Please review

File: sale_service_project/models/project.py [hunk 23]
```diff
@@ -0,0 +1,81 @@
...
+            if 'analytic_line_id' in task.material_ids._all_columns:
+                lines = lines | task.mapped('material_ids.analytic_line_id')
+            for line in lines:
+                if line.invoice_id:
+                    invoice_ids.append(line.invoice_id.id)
+                elif line.to_invoice:
+                    all_inv = False
+            task.analytic_line_ids = lines
+            task.invoice_ids = invoice_ids
+            task.invoice_exists = invoice_ids and True or False
```
Comment (yajo) [inline sale_service_project/models/project.py]: Do you mean `bool(invoice_ids)`?

File: sale_service_project/models/project.py [hunk 24]
```diff
@@ -0,0 +1,81 @@
...
+    invoice_exists = fields.Boolean(compute='_compute_analytic_line_ids')
+    is_closed = fields.Boolean(relation='stage_id.closed')
+
+    @api.multi
+    def _compute_analytic_line_ids(self):
+        for task in self:
+            all_inv = True
+            invoice_ids = []
+            lines = task.mapped('work_ids.hr_analytic_timesheet_id.line_id')
+            if 'analytic_line_id' in task.material_ids._all_columns:
```
Comment (yajo) [inline sale_service_project/models/project.py:45]: Why this? Is it not better just to depend on module `project_task_materials_stock`?

File: sale_service_project/models/project.py [hunk 25]
```diff
@@ -0,0 +1,81 @@
...
+            'context': dict(self.env.context or {}, active_ids=lines.ids),
+        }
+
+    @api.multi
+    def action_view_invoice(self):
+        result = self.env.ref('account.action_invoice_tree1').read()[0]
+        if len(self.invoice_ids) != 1:
+            result['domain'] = ("[('id', 'in', " + str(self.invoice_ids.ids) +
+                                ")]")
+        elif len(self.invoice_ids) == 1:
```
Comment (yajo) [inline sale_service_project/models/project.py]: just `else`

File: sale_service_project/wizard/product_price_service_view.xml [hunk 26]
```diff
@@ -0,0 +1,33 @@
...
+    <data>
+        
+        <record id="wiz_compute_task_price" model="ir.ui.view">
+            <field name="name">product.price.service.wizard.form</field>
+            <field name="model">product.price.service.wizard</field>
+            <field name="arch" type="xml">
+                <form string="Compute Price">
+                    <group string="Select a product to get hour price">
+                        <field name="product_id"/>
+                    </group>                    <footer>
```
Comment (yajo) [inline sale_service_project/wizard/product_price_service_view.xml]: Please put footer in another line.

**Comment by yajo** [comment]
Code reviewed, but I see lots of functionality added and no new comments in README, code or PR. Maybe this is just WIP or bug fixing, but otherwise please explain yourself when publishing changes so we reviewers know what to review.

File: sale_service_project/models/project.py (hunk 24 above)
Comment (carlosdauden) [inline sale_service_project/models/project.py:45]: It is precisely to avoid this dependence.

**Comment by carlosdauden** [comment]
At this time the module is WIP. 
What I should do for that someone with permissions adds the WIP label?

File: sale_service_project/models/project.py (hunk 23 above)
Comment (carlosdauden) [inline sale_service_project/models/project.py]: Exactly, thanks

File: sale_service_project/models/project.py (hunk 24 above)
Comment (yajo) [inline sale_service_project/models/project.py:45]: Ah ok.

**Comment by yajo** [comment]
You can also add `[WIP]` to the title. In such case, please someone remove _needs review_. Thanks for clarifying, I'll wait to review.

**Comment by rafaelbn** [comment]
Hi, thanks! Tested locally :+1: we must check travis.

File: sale_service_project/README.rst [hunk 27]
```diff
@@ -0,0 +1,72 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+
+====================
+Sale Service Project
+====================
+
```
Comment (rafaelbn) [inline sale_service_project/README.rst:8]: This module helps you to manage task created from sales order, let you choose if you wants to invoice from sale order or from task, and helps you to control with sale order must be invoiced or not depending if the task have been done or not.

File: sale_service_project/README.rst [hunk 28]
```diff
@@ -0,0 +1,72 @@
...
+This module extends the functionality of the *sale_service* module creating a
+new project for each order adding materials and works informed on the order
+line to the generated task.
+
+Usage
+=====
+
+To use this module, you need to:
+
+1. Create a service product and check 'Create Task Automatically'.
```
Comment (rafaelbn) [inline sale_service_project/README.rst]: Go to Sales -> Products and create a product with type Service, check also option 'Create task automatically'

File: sale_service_project/README.rst (hunk 12 above)
Comment (rafaelbn) [inline sale_service_project/README.rst]: Now you will see two new columns, one per works of task and one for materials which will be used in the task.

**Comment by carlosdauden** [comment]
Travis fails in sale_properties_dynamic_fields module.
Can I do something or this error isn't related with my module?

**Comment by rafaelbn** [comment]
cc @pedrobaeza

**Comment by pedrobaeza** [comment]
This PR is supposed to fix Travis: https://github.com/OCA/sale-workflow/pull/266

**Comment by rafaelbn** [comment]
OK, I merge as this module has been reviewed technically and functionally and all comments and fixed have been apply.

**Pull request merged**